test/tests/build/build_cmake_targets.vader

```diff
@@ -215,6 +224,87 @@ Execute (Get cmake targets):
+"        Assert !isdirectory("cmake-build-Release"), "Build directory shouldn't exist"
+"        silent CMake
+"        Assert filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt should be generated"
+"        let result = split(cmake4vim#CompleteBuildType(0,0,0), "\n")
+"        AssertEqual len(result), 5
+"    else
+"        echom 'SKIPPED'
+"    endif
+"
+"Execute (Override default build types):
+"    " enable tests for windows with last cmake
+"    if !has('win32') || utils#cmake#verNewerOrEq([3, 14])
+"        let g:cmake_variants = { 'Debug' : { 'cmake_build_type' : 'Debug', 'cmake_usr_args' : '-DCUSTOM_OP=ON' } }
+"        Assert !isdirectory("cmake-build-Release"), "Build directory shouldn't exist"
+"        silent CMake
+"        Assert filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt should be generated"
+"        let result = split(cmake4vim#CompleteBuildType(0,0,0), "\n")
+"        AssertEqual len(result), 4
+"        AssertEqual g:cmake_variants[ 'Debug' ][ 'cmake_usr_args' ], '-DCUSTOM_OP=ON'
+"    else
+"        echom 'SKIPPED'
+"    endif
+"
+"Execute (Store g:cmake_usr_args with default build types):
+"    " enable tests for windows with last cmake
+"    if !has('win32') || utils#cmake#verNewerOrEq([3, 14])
+"        let g:cmake_usr_args = '-DCUSTOM_OP=ON'
+"        Assert !isdirectory("cmake-build-Release"), "Build directory shouldn't exist"
+"        silent CMake
+"        Assert filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt should be generated"
+"        let result = split(cmake4vim#CompleteBuildType(0,0,0), "\n")
+"        AssertEqual len(result), 4
+"        for build_type in keys( g:cmake_variants )
+"            AssertEqual g:cmake_variants[ build_type ][ 'cmake_usr_args' ], '-DCUSTOM_OP=ON'
+"        endfor
+"    else
+"        echom 'SKIPPED'
+"    endif
+"
+"Execute (Restore default build types):
+"    " enable tests for windows with last cmake
+"    if !has('win32') || utils#cmake#verNewerOrEq([3, 14])
+"        let default_usr_args = '-DDEFAULT_OP=ON'
+"        let g:cmake_variants = { 'custom-build' : { 'cmake_build_type' : 'Debug', 'cmake_usr_args' : '-DCUSTOM_OP=ON' } }
+"        let g:cmake_usr_args = default_usr_args
+"        Assert !isdirectory("cmake-build-Release"), "Build directory shouldn't exist"
+"        silent CMake
+"        Assert filereadable("cmake-build-Release/CMakeCache.txt"), "CMakeCache.txt should be generated"
+"        let result = split( cmake4vim#CompleteBuildType(0,0,0), "\n" )
+"        AssertEqual len(result), 5
+"        silent CMakeSelectBuildType custom-build
+"        silent call RemoveDirectory("cmake-build-custom-build")
+"        AssertEqual g:cmake_usr_args, g:cmake_variants[ 'custom-build' ][ 'cmake_usr_args' ]
+"        silent CMakeSelectBuildType Release
+"        AssertEqual g:cmake_usr_args, default_usr_args
+"    else
+"        echom 'SKIPPED'
+"    endif
+
+Execute (Only default build types have default values):
```

Comment:
@ilyachur  I added only this new test and now the `MacOS cmake 2.8` test is failing, but it was passing when all 5 were added?

This seems to be the problem (it's from the log): `(6 of 40) warning:  Ignoring CMAKE_OSX_SYSROOT value: [No Name]  [Vader]  [Vader-workbench]`

Reply:
I relaunched all validation in order to be sure that it is not a sporadic. Because looks like test didn't create the build folder.

Reply:
Okay, so now all tests on master pass on all machines. The addition of this test on windows makes it run longer than the time limit of 25min.

And all tests finish on my thinkpad in 10mins, nothing hangs. I don't understand what's happening.

Reply:
I already faced with such problems in my previous PRs.

I don't know why Windows hosts hang in some cases (it always happens if some test are failed). Usually I disable or remove tests and try to find what happens.
It doesn't depend on the timeout, I had an experiment without timeout and tests were terminated after several hours.

Maybe if you cannot understand the root cause of issue and on your machine all looks good, I propose to temporary disable Windows hosts with CMake less than 3.14.
I will investigate it in the future.